From: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Date: Thu, 5 Dec 2013 18:11:23 -0500
Subject: [PATCH 17/32] qlge: Allow enable/disable rx/tx vlan acceleration
 independently
Patch-mainline: v3.13-rc4
Git-commit: 4be1028e9f99011e34dfe405d572592610776c74
References: bsc#909361 FATE#317547

o Fix the driver to allow user to enable/disable rx/tx vlan acceleration independently.

  For example:
	ethtool -K ethX rxvlan on/off
	ethtool -K ethX txvlan on/off

Signed-off-by: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Gary Ching-Pang Lin <glin@suse.com>
---
 drivers/net/ethernet/qlogic/qlge/qlge_main.c |    8 --------
 1 file changed, 8 deletions(-)

--- a/drivers/net/ethernet/qlogic/qlge/qlge_main.c
+++ b/drivers/net/ethernet/qlogic/qlge/qlge_main.c
@@ -2374,14 +2374,6 @@ static int qlge_update_hw_vlan_features(
 static u32 qlge_fix_features(struct net_device *ndev, u32 features)
 {
 	int err;
-	/*
-	 * Since there is no support for separate rx/tx vlan accel
-	 * enable/disable make sure tx flag is always in same state as rx.
-	 */
-	if (features & NETIF_F_HW_VLAN_RX)
-		features |= NETIF_F_HW_VLAN_TX;
-	else
-		features &= ~NETIF_F_HW_VLAN_TX;
 
 	/* Update the behavior of vlan accel in the adapter */
 	err = qlge_update_hw_vlan_features(ndev, features);
